Source/objects.cpp

```diff
@@ -3761,10 +3762,10 @@ void OperateShrine(int pnum, int i, int sType)
 			do {
 				r = random_(0, MAX_SPELLS);
 			} while (!(plr[pnum]._pMemSpells & SPELLBIT(r + 1)));
-			if (plr[pnum]._pSplLvl[r] >= 2)
-				plr[pnum]._pSplLvl[r] -= 2;
+			if (plr[pnum]._pSplLvl[r + 1] >= 2)
+				plr[pnum]._pSplLvl[r + 1] -= 2;

```

Comment:
actually no, look at other places where SPELLBIT and pSplLvl are used
```cpp
void MaxSpellsCheat()
{
	int i;

	for (i = 1; i < MAX_SPELLS; i++) {
		if (spelldata[i].sBookLvl != -1) {
			plr[myplr]._pMemSpells |= (__int64)1 << (i - 1);
			plr[myplr]._pSplLvl[i] = 10;
		}
	}
}
```
the index seems off by one, but that's how it should be, spell with ID 0 is SPL_NULL and isn't stored, first spell is SPL_FIREBOLT = 1 which goes into pSplLvl[0], that's why I had to do r +1 for r there

Reply:
Looks like this was some old piece of code before having SPELLBIT applied and I only looked at (i - 1) part :D LGTM

Reply:
Issues was introduced here https://github.com/diasurgical/devilution/commit/07f076d2fa927514d60eadf00d14810d45579ff7#diff-f8fa924d8b51079226adff3533bf353013bf521906c51f75798d2b3632158459R3158